lib/src/parsers/phone_parser.dart

```diff
@@ -42,6 +42,46 @@ class PhoneParser {
     return PhoneNumber(isoCode: isoCode, nsn: national);
   }
 
+  /// parses a [phoneNumber] given an [isoCode] when it is unknown whether the phone number is a local or an international one,
+  /// we just know it is specified in the country's locale
+  ///
+  /// {@macro phoneNumber}
+  ///
+  /// throws a PhoneNumberException if the isoCode is invalid
+  @internal
+  static PhoneNumber fromLocale(IsoCode isoCode, String phoneNumber) {
+    phoneNumber = TextParser.normalize(phoneNumber);
+    final metadata = MetadataFinder.getMetadataForIsoCode(isoCode);
+    // we first remove any international prefix to determine whether it's actually a local or international number
+    final withoutIntlPrefix =
+    InternationalPrefixParser.removeInternationalPrefix(
+      phoneNumber,
+      countryCode: metadata.countryCode,
+      metadata: metadata,
+    );
+    // if there was no prefix it is a national number and we proceed like in fromIsoCode
+    if (withoutIntlPrefix.length==phoneNumber.length) {
+      final withoutCountryCode = CountryCodeParser.removeCountryCode(
+          withoutIntlPrefix, metadata.countryCode);

```

Comment:
The country code removed here is the one from the locale country. This does not look right.

Reply:
You're right, plus, since we know it's not an international number it does not make sense to remove the country code at all.